src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,73 @@
+        } catch (Exception e) {
+            System.err.println("Error processing files: " + e.getMessage());
+            e.printStackTrace();
+        }
+    }
+
+    private Map<String, Integer> readData(String fileName) {
+        Map<String, Integer> operationTotals = new HashMap<>();
+        try (BufferedReader reader = new BufferedReader(new FileReader(fileName))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                String[] parts = line.split(COMMA);
+                if (parts.length == REQUIRED_PARTS_COUNT) {
+                    String operation = parts[OPERATION_INDEX].trim();
+                    try {
+                        int amount = Integer.parseInt(parts[AMOUNT_INDEX].trim());
+                        operationTotals.put(operation, operationTotals
+                                .getOrDefault(operation, DEFAULT_AMOUNT) + amount);
+                    } catch (NumberFormatException e) {
+                        System.err.println("Error parsing amount in line: " + line);
+                        e.printStackTrace();
+                    }
+                } else {
+                    System.err.println("Invalid line format: " + line);
+                }
+            }
+        } catch (IOException e) {
+            System.err.println("Can't read data from " + fileName);
+            e.printStackTrace();
+        }
+        return operationTotals;
+    }
 
+    private void writeReport(String fileName, int supplyTotal, int buyTotal, int result) {
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(fileName))) {
+            writer.write(SUPPLY + COMMA + supplyTotal);
+            writer.newLine();
+            writer.write(BUY + COMMA + buyTotal);
+            writer.newLine();
+            writer.write(RESULT + COMMA + result);
+        } catch (IOException e) {
+            System.err.println("Can't write data to " + fileName);
+            e.printStackTrace();
```

Leaving empty catch block or e.printStackTrace here is a bad practice. Better re-throw RuntimeException with original exception and informative message in the parameters:

catch (Exception e) {
    throw new RuntimeException("Can't read data from the file " + fileName, e);
}